testing/acceptance_tests/circuit_test.py

```diff
@@ -0,0 +1,744 @@
+
+
+class TestGenerateRandomAnsatzParams:
+    @pytest.mark.parametrize(
+        "number_of_layers",
+        [0, 1, 4, 7],
+    )
+    def test_generate_random_ansatz_params_using_mock_ansatz_specs(
+        self, number_of_layers
+    ):
+        # Given
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": number_of_layers,
+            "problem_size": 2,
+        }
+        seed = RNDSEED
+
+        filename = "params.json"
+        if os.path.exists(filename):
+            os.remove(filename)
+
+        # When
+        generate_random_ansatz_params(ansatz_specs=ansatz_specs, seed=seed)
+
+        # Then
+        assert os.path.exists(filename)
+        parameters = load_circuit_template_params(filename)
+        assert len(parameters) == number_of_layers
+
+        if os.path.exists(filename):
+            os.remove(filename)
+
+    def test_generate_random_ansatz_params_ansatz_specs_as_string(self):
+        # Given
+        number_of_layers = 5
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": number_of_layers,
+            "problem_size": 2,
+        }
+        seed = RNDSEED
+
+        filename = "params.json"
+        if os.path.exists(filename):
+            os.remove(filename)
+
+        # When
+        generate_random_ansatz_params(ansatz_specs=json.dumps(ansatz_specs), seed=seed)
+
+        # Then
+        assert os.path.exists(filename)
+        parameters = load_circuit_template_params(filename)
+        assert len(parameters) == number_of_layers
+
+        if os.path.exists(filename):
+            os.remove(filename)
+
+    @pytest.mark.parametrize(
+        "number_of_parameters",
+        [i for i in range(12)],
+    )
+    def test_generate_random_ansatz_params_using_number_of_parameters(
+        self,
+        number_of_parameters,
+    ):
+        # Given
+        seed = RNDSEED
+
+        filename = "params.json"
+        if os.path.exists(filename):
+            os.remove(filename)
+
+        # When
+        generate_random_ansatz_params(
+            number_of_parameters=number_of_parameters, seed=seed
+        )
+
+        # Then
+        assert os.path.exists(filename)
+        parameters = load_circuit_template_params(filename)
+        assert len(parameters) == number_of_parameters
+
+        if os.path.exists(filename):
+            os.remove(filename)
+
+    def test_generate_random_ansatz_params_fails_with_both_ansatz_specs_and_number_of_parameters(
+        self,
+    ):
+        # Given
+        number_of_parameters = 2
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": 2,
+            "problem_size": 1,
+        }
+        seed = RNDSEED
+
+        # When
```

Comment:
This is an example where annotating with given/when/then doesn't quite fit, right? How about we delete this comments from this and other tests that don't have "then"? Otherwise we can mark line with `raises` with "then" and line with call to `generate_random_ansatz_params` with "when".

Reply:
Addressed here: 0f15633